### PR TITLE
Added some other design goals/questions

### PR DESCRIPTION
**Pull request by meditans**
Some other design goals influenced by the reflex experience

### CODE REVIEW

**Comment by meditans** [comment]
Also, where should I push the answer to these questions for the reflex case?

File: CONTRIBUTIONS.md [hunk 1]
```diff
@@ -97,6 +97,9 @@ A lot of people are excited about building a JSON API in Servant and everyone se
   - Investigate the tradeoff of doing all the implementation in haskell vs. interface, via haskell, to html templates.
   - Investigate how to integrate with existing jQuery widgets (calendar, accordion, search/sort tables, editable grids, etc)
   - Forms relying only on server side validation or with mixed client/server side validation
+  - Investigate the preferred way to architecture an application. How powerful, and how general, is it?
```
Comment (saurabhnanda) [inline CONTRIBUTIONS.md:100]: How are you planning to answer this for reflex?

File: CONTRIBUTIONS.md [hunk 2]
```diff
@@ -97,6 +97,9 @@ A lot of people are excited about building a JSON API in Servant and everyone se
   - Investigate the tradeoff of doing all the implementation in haskell vs. interface, via haskell, to html templates.
   - Investigate how to integrate with existing jQuery widgets (calendar, accordion, search/sort tables, editable grids, etc)
```
Comment (saurabhnanda) [inline CONTRIBUTIONS.md:98]: Btw, we forgot this design goal. Come back to this later?

**Comment by saurabhnanda** [comment]
> Also, where should I push the answer to these questions for the reflex case?

Do something similar to #51 (markdown file in `doc` directory).

File: CONTRIBUTIONS.md (hunk 1 above)
Comment (meditans) [reply CONTRIBUTIONS.md:100]: Yes, I plan to describe all the work I did in this area (the App datatype, with one constructor for each page of the app). That also might be turned into a tiny library, but I have to get an efficiency imprimatur from Ryan first.

File: CONTRIBUTIONS.md (hunk 2 above)
Comment (meditans) [reply CONTRIBUTIONS.md:98]: I feel that the question of implementing all the markup directly in reflex vs using templates has been solved: I absolutely suggest using a quasiquoter, as we did.

The jQuery widgets probem is indeed more complex: probably it requires javascript ffi, that I'm learning only now. So I definitely don't have an answer for jQuery widgets integration.

**Pull request merged**